File: Math/1432. Max Difference You Can Get From Changing an Integer/1432. Max Difference You Can Get From Changing an Integer.py
class Solution:
    def maxDiff(self, num: int) -> int:
        num_str = str(num)

        for char in num_str:
            if char != "9":
                maxx = num_str.replace(char, "9")
                break
        else:
            maxx = num_str
        
        if num_str[0] != "1":
            minn = num_str.replace(num_str[0], "1")
        else:
            for char in num_str[1:]:
                if char not in "01":
                    minn = num_str.replace(char, "0")
                    break
            else:
                minn = num_str
        
        return int(maxx) - int(minn)

# Time: O(l)
# Space: O(l)
# Where l is the number of digits in m

File: Math/1432. Max Difference You Can Get From Changing an Integer/1432. Max Difference You Can Get From Changing an Integer.js
/**
 * @param {number} num
 * @return {number}
 */
var maxDiff = function (num) {
  const num_str = num.toString();
  let maxx, minn;

  let max_flag = false;
  for (const char of num_str) {
    if (char !== "9") {
      maxx = num_str.replaceAll(char, "9");
      max_flag = true;
      break;
    }
  }
  if (!max_flag) maxx = num_str;

  if (num_str[0] != "1") {
    minn = num_str.replaceAll(num_str[0], "1");
  } else {
    let min_flag = false;
    for (const char of num_str.slice(1)) {
      if (!"01".includes(char)) {
        minn = num_str.replaceAll(char, "0");
        min_flag = true;
        break;
      }
    }
    if (!min_flag) minn = num_str;
  }

  return Number(maxx) - Number(minn);
};

// Time: O(l)
// Space: O(l)
// Where l is the number of digits in n